src/System.Private.CoreLib/shared/System/DefaultBinder.cs

```diff
@@ -1204,6 +1204,67 @@ private static bool CreateParamOrder(int[] paramOrder, ParameterInfo[] pars, str
             return true;
         }
 
+        // CanChangePrimitive
+        // This will determine if the source can be converted to the target type
+        internal static bool CanChangePrimitive(Type source, Type target)
+        {
+            if ((source == typeof(IntPtr) && target == typeof(IntPtr)) ||
+                (source == typeof(UIntPtr) && target == typeof(UIntPtr)))
+                return true;
+
+            Primitives widerCodes = s_primitiveConversions[(int)(Type.GetTypeCode(source))];
```

Comment:
GetTypeCode returns what the type is like; not what the type actually is. For example, I can build my own type that looks like this:

```
class IntLike : IConvertible
{
    public TypeCode GetTypeCode() => TypeCode.Int32;

    ... the rest of IConvertible methods
}
```

The new and old code does not seem to behave the same way in cases like these.

Reply:
Hmm, that means the CoreRT code was already incorrect in the first place. Would something like `GetCorElementType` work for this? On some level it is exposed in CoreRT, so it would likely be possible to wire up in a way that would work for all the runtimes.

Reply:
Or maybe it makes sense to expose `TypeCode GetRealTypeCode`
or yeah let me submit a prototype based on `RuntimeTypeHandle.GetCorElementType`..

Reply:
Actually, this would be only a problem if you get the TypeCode from the instance. This gets the TypeCode from the runtime type itself, so it should not be a problem.